service_capacity_modeling/capacity_planner.py

```diff
@@ -189,13 +189,25 @@ def _set_instance_objects(
 ):
     if desires.current_clusters:
         for zonal_cluster_capacity in desires.current_clusters.zonal:
-            zonal_cluster_capacity.cluster_instance = hardware.instances[
-                zonal_cluster_capacity.cluster_instance_name
-            ]
+            if zonal_cluster_capacity.cluster_instance_name in hardware.instances:
+                zonal_cluster_capacity.cluster_instance = hardware.instances[
+                    zonal_cluster_capacity.cluster_instance_name
+                ]
+            else:
+                raise ValueError(
+                    f"Model not trained to right size cass clusters that are of "
```

Comment:
Isn't this in generic interface?

Reply:
Yes most of them are. The one's that arent we'd rather handle this much more gracefully than an index out-of-bounds error.

Reply:
Can we remove the word "cass" then?